packages/core/src/validate/run.ts

```diff
@@ -248,9 +248,19 @@ function* getStateVariableErrors(
 }
 
 function getReferencedLibraryIds(contractDef: ContractDefinition): number[] {
-  const implicitUsage = [...findAll('UsingForDirective', contractDef)].map(
-    usingForDirective => usingForDirective.libraryName.referencedDeclaration,
-  );
+  const implicitUsage = [...findAll('UsingForDirective', contractDef)]
+    .map(usingForDirective => {
+      if (usingForDirective.libraryName !== undefined) {
+        return usingForDirective.libraryName.referencedDeclaration;
+      } else if (usingForDirective.functionList !== undefined) {
+        return usingForDirective.functionList.map(func => func.function.referencedDeclaration);
```

Comment:
We don't support free functions yet (#159), so I think in this branch we should just return an empty array. (`referencedDeclaration` will refer to function ids rather than library ids.)